beacon/goclient/goclient.go

```diff
@@ -156,27 +156,72 @@ func New(
 		longTimeout = DefaultLongTimeout
 	}
 
-	httpClient, err := eth2clienthttp.New(opt.Context,
-		// WithAddress supplies the address of the beacon node, in host:port format.
-		eth2clienthttp.WithAddress(opt.BeaconNodeAddr),
-		// LogLevel supplies the level of logging to carry out.
-		eth2clienthttp.WithLogLevel(zerolog.DebugLevel),
-		eth2clienthttp.WithTimeout(commonTimeout),
-		eth2clienthttp.WithReducedMemoryUsage(true),
+	beaconAddrList := strings.Split(opt.BeaconNodeAddr, ";") // TODO: temporary using ; as separator because , is used as separator by deployment bot
+	if len(beaconAddrList) == 0 {
+		return nil, fmt.Errorf("no beacon node address provided")
+	}
+
+	var consensusClients []Client
+	var consensusClientsAsServices []eth2client.Service
+	for _, beaconAddr := range beaconAddrList {
+		httpClient, err := setupHTTPClient(opt.Context, logger, beaconAddr, commonTimeout)
+		if err != nil {
+			return nil, err
+		}
+
+		nodeVersionResp, err := httpClient.NodeVersion(opt.Context, &api.NodeVersionOpts{})
+		if err != nil {
+			logger.Error(clResponseErrMsg,
+				zap.String("api", "NodeVersion"),
+				zap.Error(err),
+			)
+			return nil, fmt.Errorf("failed to get node version: %w", err)
+		}
+		if nodeVersionResp == nil {
+			logger.Error(clNilResponseErrMsg,
+				zap.String("api", "NodeVersion"),
+			)
+			return nil, fmt.Errorf("node version response is nil")
+		}
+
+		logger.Info("consensus client connected",
+			fields.Name(httpClient.Name()),
+			fields.Address(httpClient.Address()),
+			zap.String("client", string(ParseNodeClient(nodeVersionResp.Data))),
+			zap.String("version", nodeVersionResp.Data),
+		)
+
+		consensusClients = append(consensusClients, httpClient)
+		consensusClientsAsServices = append(consensusClientsAsServices, httpClient)
+	}
+
+	err := assertSameSpec(opt.Context, consensusClients...)
```

Comment:
we should add some comments about this and the above, the conditions to make multiple bn work together (dialable, same spec, version, etc..)